tests/specs/onboarding.spec.ts

```diff
@@ -1,47 +1,144 @@
 import test from "../fixtures/common";
 import { expect } from "@playwright/test";
 import { OnboardingPage } from "../models/OnboardingPage";
+import { Modal } from "../models/Modal";
 import { DeviceAction } from "../models/DeviceAction";
 
-test("Onboarding", async ({ page }) => {
-  const onboardingPage = new OnboardingPage(page);
-  const deviceAction = new DeviceAction(page);
+const nanos = ["Nano X", "Nano S", "Blue"];
 
-  await test.step("Get started", async () => {
-    expect(await onboardingPage.getStartedButton).toBeVisible();
-    expect(await page.screenshot()).toMatchSnapshot(`getstarted.png`);
-    await onboardingPage.getStarted();
-  });
+test.describe.parallel("Onboarding", () => {
+  for (const nano of nanos) {
+    test(`[${nano}] Onboarding flow already set up`, async ({ page }) => {
+      const onboardingPage = new OnboardingPage(page);
+      const modalPage = new Modal(page);
+      const deviceAction = new DeviceAction(page);
 
-  await test.step("Terms of service", async () => {
-    expect(await page.screenshot()).toMatchSnapshot(`terms.png`);
-    await onboardingPage.acceptTerms();
-  });
+      await test.step("Get started", async () => {
+        expect(await onboardingPage.getStartedButton).toBeVisible();
+        expect(await page.screenshot()).toMatchSnapshot(`getstarted.png`);
+        await onboardingPage.getStarted();
```

Comment:
Seems a bit weird to have the expect before the actions. It's better practice to have the action first and then assert after @alalmi-ledger